src/database.cpp

```diff
@@ -213,14 +213,17 @@ bool DBResult::next()
 	return row;
 }
 
-DBInsert::DBInsert(std::string query) : query(std::move(query)) { this->length = this->query.length(); }
+DBInsert::DBInsert(std::string query, Database& db) : query(std::move(query)), db(db)
+{
+	this->length = this->query.length();
```

Comment:
`this->length` seems to always be `this->query.length()` so I think you can delete this attribute.